tee-worker/identity/client-sdk/packages/client-sdk/src/lib/requests/call-ethereum.request.ts

```diff
@@ -92,16 +93,27 @@ export async function callEthereum(
       params: [request.toHex()],
     };
 
-    const enclaveResult = await enclave.send(api, rpcRequest);
+    const [response] = await enclave.send(api, rpcRequest); // we expect 1 response only
+
+    const result: TrustedCallResult = api.createType(
+      'TrustedCallResult',
+      response.value
+    );
+
+    if (result.isErr) {
+      throw new Error(codecToString(result.asErr));
+    }
+
+    const { extrinsic_hash, block_hash } = result.asOk;
 
     return {
-      txHash,
-      response: enclaveResult,
+      response,
+      extrinsicHash: extrinsic_hash.toString(),
+      blockHash: block_hash.toString(),
```

Comment:
The `block_hash` is an `Option`, would this work?

Reply:
It won't throw, if that's what you mean. 

It'd either be: the hex value, empty strings, or `0x0000...`

but for these methods, blockHash should always be there, right? since we are watching until InBlock?

Reply:
Yes, we are waiting until the is Finalized